location_extractor/clients.py

```diff
@@ -0,0 +1,252 @@
+                        country_iso_code_lowercase text,
+                        country_name_lowercase text,
+                        subdivision_name_lowercase text,
+                        city_name_lowercase text
+                    )
+                    ''')
+
+                for col in self.index_columns:
+                    conn.execute(f'''
+                        CREATE INDEX {col} 
+                        ON locations({col}_lowercase);
+                    ''')
+
+                locations_path = os.path.join(
+                    src_dir, "data", "GeoLite2-City-CSV_20200303", "GeoLite2-City-Locations-en-processed.csv")
+
+                locations_data = []
+                with open(locations_path, 'r') as locations_file:
+                    reader = csv.reader(locations_file, delimiter=',')
+                    columns = next(reader)
+                    for i in range(len(columns) - 1):
+                        columns.append(f'{col}_lowercase')
+
+                    for row in reader:
+                        locale_code = row[0] or ''
+                        continent_code = row[1] or ''
+                        continent_name = row[2] or ''
+                        country_iso_code = row[3] or ''
+                        country_name = row[4] or ''
+                        subdivision_name = row[5] or ''
+                        city_name = row[6] or ''
+                        is_in_european_union = True if row[7] == 'True' else False
+                        locations_data.append((
+                            locale_code,
+                            continent_code,
+                            continent_name,
+                            country_iso_code,
+                            country_name,
+                            subdivision_name,
+                            city_name,
+                            is_in_european_union,
+                            locale_code.lower(),
+                            continent_code.lower(),
+                            continent_name.lower(),
+                            country_iso_code.lower(),
+                            country_name.lower(),
+                            subdivision_name.lower(),
+                            city_name.lower(),
+                        ))
+
+                conn.executemany(
+                    '''
+                        INSERT INTO locations 
+                        VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)
+                    ''',
+                    locations_data
+                )
+
+        self.default_columns = [
+            'locale_code',
+            'continent_code',
+            'continent_name',
+            'country_iso_code',
+            'country_name',
+            'subdivision_name',
+            'city_name',
+            'is_in_european_union',
+        ]
+
+        self.columns = ','.join(self.default_columns)
+
+    def connect(self):
+        return sqlite3.connect(self.dbpath)
```

Comment:
what's your opinion about creating `connection` property instead of this method?

```suggestion
    @property
    def connection(self) -> sqlite3.Connection:
        return sqlite3.connect(self.dbpath)
```